hack/update-bazel.sh fails with several complaints about "too few values in struct initializer"

<!-- This form is for bug reports and feature requests ONLY! 

If you're looking for help check [Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes) and the [troubleshooting guide](https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/).
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature


**What happened**:
```
$ ./hack/update-bazel.sh 
Cloning into '/tmp/tmp.TJKUY3llxw/go/src/github.com/kubernetes/repo-infra'...
remote: Counting objects: 848, done.
remote: Total 848 (delta 0), reused 0 (delta 0), pack-reused 848
Receiving objects: 100% (848/848), 261.63 KiB | 6.38 MiB/s, done.
Resolving deltas: 100% (385/385), done.
Cloning into '/tmp/tmp.TJKUY3llxw/go/src/github.com/bazelbuild/rules_go'...
remote: Counting objects: 9243, done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 9243 (delta 0), reused 0 (delta 0), pack-reused 9237
Receiving objects: 100% (9243/9243), 2.08 MiB | 8.25 MiB/s, done.
Resolving deltas: 100% (5975/5975), done.
# github.com/bazelbuild/rules_go/go/tools/gazelle/merger
go/tools/gazelle/merger/fix.go:93:28: too few values in struct initializer
go/tools/gazelle/merger/merger.go:498:19: too few values in struct initializer
go/tools/gazelle/merger/merger.go:502:19: too few values in struct initializer
# github.com/bazelbuild/rules_go/go/tools/gazelle/rules
go/tools/gazelle/rules/generator.go:196:16: too few values in struct initializer
go/tools/gazelle/rules/sort_labels.go:30:16: too few values in struct initializer
!!! [0226 14:06:43] Call tree:
!!! [0226 14:06:43]  1: ./hack/update-bazel.sh:34 kube::util::go_install_from_commit(...)
!!! Error in /home/mspreitz/go/src/k8s.io/kubernetes/hack/lib/util.sh:487
  Error in /home/mspreitz/go/src/k8s.io/kubernetes/hack/lib/util.sh:487. '((i<3-1))' exited with status 2
Call stack:
  1: /home/mspreitz/go/src/k8s.io/kubernetes/hack/lib/util.sh:487 kube::util::go_install_from_commit(...)
  2: ./hack/update-bazel.sh:34 main(...)
Exiting with status 1
!!! [0226 14:06:43] Call tree:
!!! [0226 14:06:43]  1: ./hack/update-bazel.sh:34 kube::util::go_install_from_commit(...)
!!! Error in /home/mspreitz/go/src/k8s.io/kubernetes/hack/lib/util.sh:472
  Error in /home/mspreitz/go/src/k8s.io/kubernetes/hack/lib/util.sh:472. '((i<3-1))' exited with status 1
Call stack:
  1: /home/mspreitz/go/src/k8s.io/kubernetes/hack/lib/util.sh:472 kube::util::go_install_from_commit(...)
  2: ./hack/update-bazel.sh:34 main(...)
Exiting with status 1
```

**What you expected to happen**:
I expected the update to run successfully.

**How to reproduce it (as minimally and precisely as possible)**:
Checkout.  `build/run.sh make`.  `hack/update-bazel.sh`

**Anything else we need to know?**:
I get this in both master and release-1.9.  I tested most carefully in release-1.9 (because I am working on a cherry-pick right now).  In particular, I blew away the container and image I had used earlier for building.  Before running `hack/update-bazel.sh` I established this state:
```
mspreitz@ubu17dev:~/go/src/k8s.io/kubernetes$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
hello-world         latest              f2a91732366c        3 months ago        1.85kB
mspreitz@ubu17dev:~/go/src/k8s.io/kubernetes$ docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
mspreitz@ubu17dev:~/go/src/k8s.io/kubernetes$ 
```

Then I ran `build/run.sh make` --- which took a while but eventually completed.  It left me this state:
```
mspreitz@ubu17dev:~/go/src/k8s.io/kubernetes$ docker ps -a
CONTAINER ID        IMAGE                                    COMMAND                  CREATED             STATUS                      PORTS               NAMES
dae3627e1150        kube-build:build-da2f51b002-5-v1.9.2-1   "chown -R 1000:1000 …"   20 minutes ago      Exited (0) 17 minutes ago                       kube-build-data-da2f51b002-5-v1.9.2-1
mspreitz@ubu17dev:~/go/src/k8s.io/kubernetes$ docker images
REPOSITORY                            TAG                           IMAGE ID            CREATED             SIZE
kube-build                            build-da2f51b002-5-v1.9.2-1   b90e5435889f        20 minutes ago      2.18GB
hello-world                           latest                        f2a91732366c        3 months ago        1.85kB
gcr.io/google_containers/kube-cross   v1.9.2-1                      ba98259b52f7        3 months ago        1.74GB
```

After that is when I ran the `hack/update-bazel.sh` that is exhibited at the start of this issue.

**Environment**:
- Kubernetes version (use `kubectl version`): Commit 77487e4d2e8ebfddd2539544d806f3a17ea61834 (current release-1.9)
- Cloud provider or hardware configuration: local VM
- OS (e.g. from /etc/os-release): Ubuntu 17.10
- Kernel (e.g. `uname -a`): Linux ubu17dev 4.13.0-21-generic #24-Ubuntu SMP Mon Dec 18 17:29:16 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Others:
Docker 17.12.0-ce

--- a/build/root/BUILD.root
+++ b/build/root/BUILD.root
@@ -2,6 +2,7 @@
 # gazelle:exclude _gopath
 # gazelle:exclude _output
 # gazelle:exclude _tmp
+# gazelle:prefix k8s.io/kubernetes
 
 package(default_visibility = ["//visibility:public"])
 

--- a/hack/update-bazel.sh
+++ b/hack/update-bazel.sh
@@ -32,7 +32,7 @@ kube::util::go_install_from_commit \
     97099dccc8807e9159dc28f374a8f0602cab07e1
 kube::util::go_install_from_commit \
     github.com/bazelbuild/bazel-gazelle/cmd/gazelle \
-    39e8257565d3917c1674bf37d6538c10cdac2260
+    a85b63b06c2e0c75931e57c4a1a18d4e566bb6f4
 
 touch "${KUBE_ROOT}/vendor/BUILD"
 

